doc/contributing.md

```diff
@@ -0,0 +1,83 @@
+If the pull request does not have a clear purpose and benefits it will not be accepted. 
+
+* **The pull request is documented** - The HDF5 developers must understand not only *what* a change is doing, but *how* it is doing it.  Documenting the code makes it easier for us to understand your patch and will help to maintaine the code in the future. 
+
+* **The pull request passes HDF5 regression testing** - Any issue fixed or functionality added should be accompanied by the corresponding tests and pass HDF5 regression testing run by The HDF Group. We do not expect you to perform comprehensive testing across a multiple platforms before we accept the pull request. If the pull request does not pass regression testing after the merge, The HDF Group developers will work with you on the fixes. 
+
+* **The pull request does not compromise the principles behind HDF5** - HDF5 has a 100% commitment to backward compatibility.  
+	* Any file ever created with HDF5 must be readable by any future version of HDF5.
+   If the purpose of your patch  is to modify HDF5 data model or file format,
+ **please** discuss this with us first. File format changes and features required those changes can be introduced only in a new major release. 
+	* HDF5 has a commitment to remaining *machine-independent*; data created on one platform/environment/architecture **must** remain readable by HDF5 on any other. 
+	* For binary compatibility no changes are allowed to public APIs and data structures in the maintenance releases; new APIs can be added.
+
+* **New features are documented** - Any new features should have proper documentation; talk to us if you have any questions.
+
+
+# Checklist <A NAME="checklist"></A>
+
+Please make sure that you check the items applicable to your pull request:
+
+* Code 
+  * [ ] Does the pull request have a coresponding GitHub issue and clear purpose?
+  * [ ] Does the pull request follow HDF5 best practices (naming convensions, code portability, code structure, etc.)? <<TODO: link to the document>>
+  * [ ] If changes were done to autotools build were they added to CMake and vice versa??
+  * [ ] Is the pull request applicable to any other branches? If yes, which ones? Please document it in the GitHub issue.
+  * [ ] Is the new code sufficiently documented for future maintenance?
+* Documentation
+  * [ ] Was the change described in the release_docs/RELEASE.txt file?
+  * [ ] Was MANIFEST updated if new files had been added to the source?
+  * [ ] Was new function documented in the corresponding public header file using Doxygen? <<TODO: link to Doxygen instructions>>
+  * [ ] Was new functionality documented for the HDF5 community (the level of documentation depends on the feature; ask us what would be appropriate)
+* Testing
+  * [ ] Does the pull request have tests?
+  * [ ] If a new feature is added, does it affect HDF5 library perfromance?
+  * [ ] Does new feature introduce backward or forward incompatibility? <<TODO: link to the document>>
```

Comment:
maybe include, provide, or address instead of introduce

Reply:
Reworked the line and moved it to coding section. We do have info on Macros, but not on  format versioning. I need to get things together and post them.